### PR TITLE
Enable Windows build

### PR DESCRIPTION
**Pull request by wshanks**
This is just seeing what happens with trial and error for now.

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by wshanks** [comment]
Currently this is stuck on `compiler is out of heap space` (for example [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=776706&view=logs&j=5be07ae1-d8ba-5406-47b6-8e3a3a12f825&t=0bf03e01-0bec-5b85-5316-b1633322e895) from d8b14ed). Upstream builds in GitHub Actions without error ([most recent build log](https://github.com/Qiskit/qiskit-aer/actions/runs/6110719184/job/16584606646) at this time). Looking at the resources listed for [Azure Pipelines](https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#hardware) and [GitHub Actions](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources), I see that both should have 2 CPUs and 7 GB of RAM. So the current possibilities that I can think of are:

* Upstream manages to compile just under the 7 GB limit and something about the conda tooling pushes it over the limit. I am not sure what to do if this is the case.
* Something about the way I have set up the build is leading to excessive memory usage. I am not sure how to diagnose this, but it would be more hopeful than the previous case.

I looked closely at the build logs mentioned above and I didn't see much difference between here and upstream besides using Visual Studio 17 2022 here and Visual Studio 16 2019 there.

**Comment by jaimergp** [comment]
Maybe you can increase the pagefile. I don't know if it's documented (we should!), but you can find [this block](https://github.com/conda-forge/numpy-feedstock/blob/14809f107a7d8c834232f3a45634d80a88b289f0/conda-forge.yml#L1-L4) in the `numpy-feedstock`:

```yaml
azure:
  settings_win:
    variables:
      SET_PAGEFILE: 'True'
```

Add it and rerender to apply it to the pipelines. Let's see 🤞

**Comment by wshanks** [comment]
Hmm, the build recipe step took two minutes longer, so maybe the pagefile change did something, but the build still ran into the compiler out of heap space message. I am not that familiar with building on Windows. I feel like I must be missing something in the set up because I don't think this should be an excessively large build.

**Comment by jaimergp** [comment]
Oh nice, now it does build (there's a missing runtime dep but that should be easy to fix). So the 2019 works but 2022 doesn't? That's interesting.

**Comment by wshanks** [comment]
> So the 2019 works but 2022 doesn't? That's interesting.

Yes, I am not sure why that is, but upstream builds with windows-2019, so I will hope upstream will move to windows-2022 and figure it out before windows-2019 can not be used for conda-forge any more.

> there's a missing runtime dep but that should be easy to fix

It's actually that the cmake configuration [hardcodes a bundled openblas](https://github.com/Qiskit/qiskit-aer/blob/39487dbf8cfe002dbf50cbadd923609c933a4a30/CMakeLists.txt#L228-L249) unless you specify a path to the blas library. I don't think either of those options are viable for conda-forge, so we need to patch in and hopefully get upstream to accept an option not to use the bundled blas.

**Comment by wshanks** [comment]
It builds now! I included a patch to let cmake use the conda openblas library. I submitted the patch [here](https://github.com/Qiskit/qiskit-aer/pull/1940). If a form of the patch is accepted upstream, I will backport it into the recipe here and release a Windows build (with the intent to remove the patch once it is included in a release).

**Comment by wshanks** [comment]
@leofang Sorry if I asked you about this already, would you worry about things like this (this one gets warnings on Linux as well)?

```
WARNING (qiskit-aer): run-exports library package conda-forge::spdlog-1.12.0-h64d2f7d_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

File: recipe/bld.bat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SET CMAKE_ARGS="%CMAKE_ARGS% -DUSE_BUNDLED_BLAS_WIN:BOOL=FALSE"
```
Comment (jaimergp) [inline recipe/bld.bat]:
```suggestion
SET "CMAKE_ARGS=%CMAKE_ARGS% -DUSE_BUNDLED_BLAS_WIN:BOOL=FALSE"
```

CMD quoting rules are tricky. This is a bit more robust, but you might see issues with nested quotes. Keep an eye.

Comment (wshanks) [reply recipe/bld.bat]: I had nested quotes with the version you commented on, which did not work, but then it worked with no quotes. I will try your suggestion.

Comment (wshanks) [reply recipe/bld.bat]: By the way, is this how you set `CMAKE_ARGS` on Windows when not invoking `cmake` directly? scikit-build's documentation suggests using `python setup.py ... <cmake args>` but I want to keep `pip install` as more future-proof (not sure if `python setup.py` still works any way). The alternative way to set CMake options is to use `CMAKE_ARGS` but conda-forge's build set up overrides `CMAKE_ARGS` when I try to set it in the recipe, so I moved it down in the script like this, so I could extend what conda-forge set after it set it.

Comment (jaimergp) [reply recipe/bld.bat]: Yes, conda-forge infra sets `CMAKE_ARGS` for you. It looks like the current approach works so... yay? 🥳

Comment (wshanks) [reply recipe/bld.bat]: Yes, it's good now :slightly_smiling_face: I just wasn't sure if dropping to bld.bat and extending `CMAKE_ARGS` was idiomatic for conda-forge or if there were some other way to extend `CMAKE_ARGS`.

**Pull request merged**